haskell/cabal.bzl

```diff
@@ -719,6 +719,25 @@ def _invert(d):
     """Invert a dictionary."""
     return dict(zip(d.values(), d.keys()))
 
+def _from_string_keyed_label_list_dict(d):
```

Comment:
Nit: It's probably worth mentioning here that this function and the following ones exist only to work around https://github.com/bazelbuild/bazel/issues/7989

Reply:
Good point, there's a comment down in `def stack_snapshot`, but it doesn't hurt to have it in these functions, too.